splash/browser_tab.py

```diff
@@ -870,6 +871,72 @@ def _post_mouse_event(self, type, button, x, y):
         event = QMouseEvent(type, point, q_button, buttons, modifiers)
         QApplication.postEvent(self.web_page, event)
 
+    def send_text(self, text):
+        """
+        Send full text as a sole key event to webpage
+        :param text string to be sent as input
+        :return: None
+        """
+        return self._send_text(text, key_type=0)
+
+    def send_keys(self, text):
+        """
+        Send key events to webpage
+        :param text string to be sent as key strokes
+        :return: None
+        """
+        for word in text.split():
+            self._send_word(word)
+
+    EDMACRO_SHORTCUTS = {
+        'RET': Qt.Key_Return,
+        'SPC': Qt.Key_Space,
+        'TAB': Qt.Key_Tab,
+        'DEL': Qt.Key_Delete,
+        'ESC': Qt.Key_Escape,
+        # 'LFD': 0, ## QChar.LineFeed?
+        # 'NUL': 0, ## QChar.Null?
+    }
+
+    QT_KEY_INPUTS = {
+        # TODO: More needed?
+        Qt.Key_Return: '\n',
+        Qt.Key_Enter: '\n',
+        Qt.Key_Space: ' ',
+        Qt.Key_Tab: '\t',
+    }
+
+    def _send_word(self, word):
```

Comment:
Why is it called `_send_word`? 
I think it should work better as a separate function in splash.qtutils.